### PR TITLE
Title json

### LINKED ISSUE #43: add a --name (or --title) swith
The name (or title) switch should be emitted in the reaults. We may even add the time at which it finished. This makes the json results easier to debug.

### PR DESCRIPTION
**Pull request by thekemkid**
fixes #43 

Also found an issue where the timeout flag was missing.

### CODE REVIEW

**Comment by mcollina** [comment]
You know I'm asking you to add a unit test :)

**Comment by thekemkid** [comment]
I added some assertions, not sure how else I could test that? Are those okay? :)

File: lib/run.js [hunk 1]
```diff
@@ -139,12 +139,15 @@ function run (opts, cb) {
...
       let result = {
+        title: opts.title,
         requests: histAsObj(requests, totalRequests),
         latency: addPercentiles(latencies, histAsObj(latencies)),
         throughput: histAsObj(throughput, totalBytes),
         errors: errors,
         timeouts: timeouts,
         duration: Math.round((Date.now() - startTime) / 1000),
+        start: startTime,
+        finish: Date.now(),
```
Comment (mcollina) [inline lib/run.js]: I think we should have those as Date objects, rather than as epocs. Easier to print :).

Comment (thekemkid) [inline lib/run.js]: Epochs are easier for transmission/serialisation though, right? I'll create them as objects though if you want.

Comment (mcollina) [inline lib/run.js]: please do, those output will very likely be consumed by humans, and `Date` do jsonify.

Comment (thekemkid) [inline lib/run.js]: done

Comment (mcollina) [inline lib/run.js]: `start` is still an epoch, you will need to convert it to a `Date` object: https://github.com/mcollina/autocannon/blob/master/lib/run.js#L79

Comment (thekemkid) [inline lib/run.js]: startTime needs to remain an epoch for the duration calculation, but I use new Date(startTime) to convert back to an object from the epoch. Do you want me to convert startTime to an object and then convert it to an epoch for the duration calculation? it's just two ways of achieving the same thing.

File: lib/run.js [hunk 2]
```diff
@@ -139,12 +139,15 @@ function run (opts, cb) {
...
       let result = {
+        title: opts.title,
         requests: histAsObj(requests, totalRequests),
         latency: addPercentiles(latencies, histAsObj(latencies)),
         throughput: histAsObj(throughput, totalBytes),
         errors: errors,
         timeouts: timeouts,
         duration: Math.round((Date.now() - startTime) / 1000),
+        start: new Date(startTime),
+        finish: new Date(),
```
Comment (thekemkid) [inline lib/run.js]: @mcollina Here. I'm using `new Date(value)`: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Date

Comment (mcollina) [inline lib/run.js]: oh I didn't see that. All good for me!

Comment (thekemkid) [inline lib/run.js]: No problem, we were still talking on the outdated diff 👍

**Pull request merged**